docker/types/services.py

```diff
@@ -319,6 +323,13 @@ def __init__(self, cpu_limit=None, mem_limit=None, cpu_reservation=None,
             reservation['NanoCPUs'] = cpu_reservation
         if mem_reservation is not None:
             reservation['MemoryBytes'] = mem_reservation
+        if generic_resources is not None:
+            if not isinstance(generic_resources, list):
```

Comment:
Oops! I wasn't super clear in what I meant, sorry!
I actually think we can have both notations, the `dict` form that you implemented originally with the list form as a fallback. See for example https://github.com/docker/docker-py/blob/master/docker/types/containers.py#L242-L246